### PR TITLE
OS Script templates improvements

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
Refactoring of OS scripts:
* unify the installation and configuration across different OSes
* flatcar improvements
* flatcar containerd config generation (a-la we do it in machine-controller)

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
OS Script templates improvements
```

### CODE REVIEW

File: pkg/scripts/render.go [hunk 1]
```diff
@@ -210,39 +207,49 @@ var (
...
-			EOF
+			{{ template "container-runtime-daemon-config" . }}
+			{{ template "containerd-systemd-environment" . }}
 
-			sudo mkdir -p /etc/systemd/system/containerd.service.d
-			cat <<EOF | sudo tee /etc/systemd/system/containerd.service.d/environment.conf
+			cat <<EOF | sudo tee /etc/systemd/system/containerd.service.d/10-kubeone.conf
 			[Service]
 			Restart=always
-			EnvironmentFile=-/etc/environment
```
Comment (xmudrii) [inline pkg/scripts/render.go:238]: Why have you removed this? Isn't this used in the proxy setups?

Comment (kron4eg) [reply pkg/scripts/render.go:238]: it's extracted in individual (but shared) template where environment.conf is set

File: pkg/scripts/testdata/TestKubeadmFlatcar-force.golden [hunk 2]
```diff
@@ -74,11 +74,17 @@ cat <<EOF | sudo tee /etc/docker/daemon.json
...
 cat <<EOF | sudo tee /etc/crictl.yaml
 runtime-endpoint: unix:///var/run/dockershim.sock
 EOF
 
+sudo mkdir -p /etc/systemd/system/containerd.service.d
+cat <<EOF | sudo tee /etc/systemd/system/containerd.service.d/environment.conf
+[Service]
+Restart=always
+EnvironmentFile=-/etc/environment
+EOF
```
[the remark this answers is not in the document]
Comment (kron4eg) [reply pkg/scripts/testdata/TestKubeadmFlatcar-force.golden:86]: no, I don't think so. why would it?

File: pkg/scripts/testdata/TestKubeadmDebian-with_containerd.golden [hunk 3]
```diff
@@ -96,22 +86,23 @@ SystemdCgroup = true
...
-sudo mkdir -p /etc/systemd/system/containerd.service.d
-cat <<EOF | sudo tee /etc/systemd/system/containerd.service.d/environment.conf
-[Service]
-Restart=always
-EnvironmentFile=-/etc/environment
-EOF
 
-sudo systemctl daemon-reload
-sudo systemctl enable --now containerd
-sudo systemctl restart containerd
```
[the remark this answers is not in the document]
Comment (kron4eg) [reply pkg/scripts/testdata/TestKubeadmDebian-with_containerd.golden]: Good catch, I've forgot about it in deb script

File: pkg/scripts/testdata/TestKubeadmFlatcar-force.golden (hunk 2 above)
[the remark this answers is not in the document]
Comment (xmudrii) [reply pkg/scripts/testdata/TestKubeadmFlatcar-force.golden:86]: Looks like I left a comment in the wrong place. Please ignore it.

**Review by xmudrii — APPROVED** [review]

**Pull request merged**